### PR TITLE
CORTX-30863: fix: Moved num_keys save operation from Conf to MappedConf

### PR DESCRIPTION
**Pull request by kailashyadav11**
Signed-off-by: Kailash Yadav <kailash.yadav@seagate.com>

# Problem Statement
- Mini provisioning is failing in motr due to add_num_keys api in confstore had removal of Conf.save method.

# Design
-  [For Bug describe the fix here. ](https://jts.seagate.com/browse/CORTX-30863)
- Moving the .save() call to MappedConf class should fix the issue
-  For Feature, Post the link to the solution page on the confluence CORTX Foundation Library 

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]:  Y

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]: Y
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]: N
- Confirm that Test Cases are added for new features added [Y/N]:  Y
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: Y
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]:  Y

# Review Checklist 
####  Before posting the PR please ensure:
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] New package/s added to setup.py?
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [x] Verification needs to be done before marked as Closed/Verified.

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:
- [x] ConfStore
- [ ] KVStore
- [ ] ConfCli
- [ ] Test cases added for all above 3
- [ ] changes done in all the KVpayloads (ConsulKvPayload, IniKvPayload, KvPayload)

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Comment by kailashyadav11** [comment]
Unit tests:

[root@ssc-vm-rhev4-2199 conf_store]# python3 test_conf_store.py
........................................................
----------------------------------------------------------------------
Ran 56 tests in 0.222s

OK

File: py-utils/src/utils/conf_store/conf_store.py [hunk 1]
```diff
@@ -390,8 +390,9 @@ def search(self, parent_key: str, search_key: str, value: str = None):
         return Conf.search(self._conf_idx, parent_key, search_key, value)
 
     def add_num_keys(self):
-        """Add "num_xxx" keys for all the list items in ine KV Store."""
+        """Add "num_xxx" keys for all the list items in ine Cortx confstore."""
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/conf_store/conf_store.py:393]: Please remove 'Cortx' keyword.

Comment (kailashyadav11) [reply py-utils/src/utils/conf_store/conf_store.py]: done

File: py-utils/src/utils/conf_store/conf_store.py [hunk 2]
```diff
@@ -390,8 +390,9 @@ def search(self, parent_key: str, search_key: str, value: str = None):
...
         Conf.add_num_keys(self._conf_idx)
+        Conf.save(self._conf_idx)
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_store.py:395]: All operations (add_num_keys, set, delete) are reflected in memory only. Caller needs to call explicit save() for it to be reflected on disk.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_store.py (hunk 2 above)
Comment (kailashyadav11) [reply py-utils/src/utils/conf_store/conf_store.py:395]: @ujjwalpl I see there is a class named MappedConf (Config Store with fixed target) where 'set' methods have the save operation save() called in them and hence this was added in the same class. 

If this is not expected I will make the save() call from caller end(provisioner).

**Comment by kailashyadav11** [comment]
All operations (add_num_keys, set, delete) are reflected in memory only. Caller needs to call explicit save() for it to be reflected on disk.,

**Pull request closed without merging**